### PR TITLE
Upgrade sigma to v5.0.7

### PR DESCRIPTION
**Pull request by aslesarenko**
In this PR:
- sigma dependency updated
- com.google.guava dependency explicitly added
- code migrated

### CODE REVIEW

File: ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/ExtendedPublicKey.scala [hunk 1]
```diff
@@ -1,13 +1,15 @@
 package org.ergoplatform.wallet.secrets
 
-import java.util
+import org.bouncycastle.math.ec.ECPoint
```
Comment (kushti) [inline ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/ExtendedPublicKey.scala:3]: unused import

File: ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/ExtendedPublicKey.scala [hunk 2]
```diff
@@ -56,11 +58,15 @@ object ExtendedPublicKey {
...
+    val childKey = CryptoFacade.multiplyPoints(
+      DLogProverInput(childKeyProtoDecoded).publicImage.value,
+      parentKey.key.value)
+    if (childKeyProtoDecoded.compareTo(CryptoConstants.groupOrder) >= 0 || CryptoFacade.isInfinityPoint(childKey)) {
       deriveChildPublicKey(parentKey, idx + 1)
     } else {
-      new ExtendedPublicKey(childKey.getEncoded(true), childChainCode, parentKey.path.extended(idx))
+      new ExtendedPublicKey(
+        CryptoFacade.encodePoint(childKey, true),
+        childChainCode, parentKey.path.extended(idx))
```
Comment (kushti) [inline ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/ExtendedPublicKey.scala:69]: formatting is weird

File: ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/ExtendedSecretKey.scala [hunk 3]
```diff
@@ -1,15 +1,17 @@
 package org.ergoplatform.wallet.secrets
 
+import org.bouncycastle.math.ec.ECPoint
```
Comment (kushti) [inline ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/ExtendedSecretKey.scala:3]: unused import

File: ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/ExtendedSecretKey.scala [hunk 4]
```diff
@@ -30,8 +32,11 @@ final class ExtendedSecretKey(private[secrets] val keyBytes: Array[Byte],
 
   def child(idx: Int): ExtendedSecretKey = ExtendedSecretKey.deriveChildSecretKey(this, idx)
 
+  /** Returns extended public key corresponding to this secret key. */
   def publicKey: ExtendedPublicKey =
-    new ExtendedPublicKey(privateInput.publicImage.value.getEncoded(true), chainCode, path.toPublicBranch)
+    new ExtendedPublicKey(
+      CryptoFacade.encodePoint(privateInput.publicImage.value, compressed = true),
+      chainCode, path.toPublicBranch)
```
Comment (kushti) [inline ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/ExtendedSecretKey.scala:39]: weird formatting

**Review by kushti — APPROVED** [review]
Approved, but exponentiatePoint / multiplyPoints operations look confusing, made an issue 

https://github.com/ScorexFoundation/sigmastate-interpreter/issues/876

Also please check comments

File: ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/ExtendedPublicKey.scala (hunk 1 above)
Comment (aslesarenko) [reply ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/ExtendedPublicKey.scala:3]: fixed

File: ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/ExtendedPublicKey.scala (hunk 2 above)
Comment (aslesarenko) [reply ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/ExtendedPublicKey.scala:69]: fixed

File: ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/ExtendedSecretKey.scala (hunk 4 above)
Comment (aslesarenko) [reply ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/ExtendedSecretKey.scala:39]: fixed

**Pull request merged**